luarules/gadgets/game_unit_market.lua

```diff
@@ -149,19 +149,37 @@ local function offerUnitForSale(unitID, sale_price, msgFromTeamID)
     end
 end
 
+local t2cons = {"armack", "armacv", "armaca", "armacsub", "corack", "coracv", "coraca", "coracsub"}
+function contains(table, value)
+    for _, v in pairs(table) do
+        if v == value then return true end
+    end
+    return false
+end
+
+-- Override unit sharing block from other modoptions
 local disable_unit_sharing = (
     Spring.GetModOptions().disable_unit_sharing
  or (Spring.GetModOptions().tax_resource_sharing_amount or 0) ~= 0)
 and Spring.GetModOptions().unit_market
 local saleWhitelist = {}
 
 local function tryToBuyUnit(unitID, msgFromTeamID)
+
     if not unitID or unitsForSale[unitID] == nil or unitsForSale[unitID] == 0 then return end
     local unitDefID = spGetUnitDefID(unitID)
     if not unitDefID then return end
     local unitDef = UnitDefs[unitDefID]
     if not unitDef then return end
 
+    
+    -- Only allow selling t2 cons when unit sharing is disabled
```

Comment:
Limiting it to only selling t2 seems counterintuitive?

Reply:
I'm using the unit market to implement an exception -- that players are used to buying t2. In general, buying 'anything' is a form of boosting, since you save the E cost, BP, and time.

By default tax unit sharing will only disable sharing of economy/labs. You are free to share combat units or other stuff without payment.

Reply:
For context, this is a competitive modoption for a small subset of competitive players. The ruleset has to cover everything to achieve the desired effect (also a similiar ruleset hasn't been tested before). I'm open for ways to make the option selection more explanatory.
For ruleset changes maybe discuss https://discord.com/channels/549281623154229250/1253060706039890000

Tax Resource Sharing can be explained as: "You can't share eco/production, can't assist allied labs, and resource sharing is taxed. You can buy/sell t2 at normal price via unit Market".